### PR TITLE
feat(sdk-coin-avaxp): implement is wallet address method

### PR DESCRIPTION
**Pull request by jrafael-simtlix**
STLX-18432

### CODE REVIEW

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 1]
```diff
@@ -83,18 +85,26 @@ export class AvaxP extends BaseCoin {
     return true;
   }
 
-  verifyAddress(params: VerifyAddressOptions): boolean {
```
Comment (jrafael-simtlix) [inline modules/sdk-coin-avaxp/src/avaxp.ts:86]: deprecated use isWalletAddress instead

https://github.com/BitGo/BitGoJS/blob/3e60cef71a0ae76b378356508338738eac49a920/modules/sdk-core/src/bitgo/baseCoin/baseCoin.ts#L240-L245

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts [hunk 2]
```diff
@@ -340,18 +341,148 @@ describe('Avaxp', function () {
     });
   });
 
-  describe('Validation', function () {
+  describe('Address Validation', function () {
+    const keychains = [
```
Comment (jicli) [inline modules/sdk-coin-avaxp/test/unit/avaxp.ts:345]: lets add them to `TestData`

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts [hunk 3]
```diff
@@ -340,18 +341,148 @@ describe('Avaxp', function () {
...
       const validAddress = 'P-fuji15jamwukfqkwhe8z26tjqxejtjd3jk9vj4kmxwa';
       basecoin.isValidAddress(validAddress).should.be.true();
     });
 
     it('should fail to validate invalid address', function () {
-      const invalidAddresses = ['asdadsaaf', 'fuji15jamwukfqkwhe8z26tjqxejtjd3jk9vj4kmxwa'];
+      const invalidAddresses = ['', 'asdadsaaf', '15x3z4rvk8e7vwa6g9lkyg89v5dwknp44858uex'];
 
       for (const address of invalidAddresses) {
+        should.doesNotThrow(() => basecoin.isValidAddress(address));
```
Comment (jicli) [inline modules/sdk-coin-avaxp/test/unit/avaxp.ts:383]: why should it not throw?

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts [hunk 4]
```diff
@@ -340,18 +341,148 @@ describe('Avaxp', function () {
...
         basecoin.isValidAddress(address).should.be.false();
       }
     });
+
+    it('should validate address', function () {
+      const validAddresses = [
+        'P-fuji15x3z4rvk8e7vwa6g9lkyg89v5dwknp44858uex',
+        'P-avax143q8lsy3y4ke9d6zeltre8u2ateed6uk9ka0nu',
```
Comment (jicli) [inline modules/sdk-coin-avaxp/test/unit/avaxp.ts:391]: these actually not legit, one is fuji and one is avax

**Review by jrafael-simtlix — COMMENTED** [review]
@jicli I done those changes.

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 5]
```diff
@@ -83,18 +85,26 @@ export class AvaxP extends BaseCoin {
...
-    const bintools = BinTools.getInstance();
-    const address = bintools.parseAddress(params.address, 'P');
-    return address !== undefined;
-  }
-
   /**
    * Check if address is valid, then make sure it matches the root address.
    *
    * @param {VerifyAddressOptions} params address and rootAddress to verify
```
Comment (CapnMigraine) [inline modules/sdk-coin-avaxp/src/avaxp.ts:91]: nit: avaxp doesn't use root address, let's rename to unlock address

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 6]
```diff
@@ -83,18 +85,26 @@ export class AvaxP extends BaseCoin {
...
+      throw new InvalidAddressError(`invalid address: ${address}`);
+    }
+    if (!keychains || keychains.length !== 3) {
+      throw new Error('Invalid keychains');
+    }
+    const walletAddress = keychains.map((keychain) =>
+      new AvaxpLib.KeyPair({ pub: keychain.pub }).getAddress(this._staticsCoin.network.type)
+    );
+
+    if (!walletAddress.some((add) => add.endsWith(address))) {
```
Comment (CapnMigraine) [inline modules/sdk-coin-avaxp/src/avaxp.ts:105]: why is `endsWith` needed here? shouldn't they be equal?

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 7]
```diff
@@ -83,18 +85,26 @@ export class AvaxP extends BaseCoin {
...
    */
   isWalletAddress(params: VerifyAddressOptions): boolean {
+    const { address, keychains } = params;
+    if (!this.isValidAddress(address)) {
+      throw new InvalidAddressError(`invalid address: ${address}`);
+    }
+    if (!keychains || keychains.length !== 3) {
+      throw new Error('Invalid keychains');
+    }
+    const walletAddress = keychains.map((keychain) =>
```
Comment (CapnMigraine) [inline modules/sdk-coin-avaxp/src/avaxp.ts:101]: nit: rename
```suggestion
    const unlockAddresses = keychains.map((keychain) =>
```

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts [hunk 8]
```diff
@@ -340,18 +342,63 @@ describe('Avaxp', function () {
     });
   });
 
-  describe('Validation', function () {
+  describe('Address Validation', function () {
```
Comment (CapnMigraine) [inline modules/sdk-coin-avaxp/test/unit/avaxp.ts:345]: nit: we validate more than just addresses inside this block
```suggestion
  describe('Validation', function () {
```

**Review by CapnMigraine — CHANGES_REQUESTED** [review]

File: modules/sdk-coin-avaxp/src/avaxp.ts (hunk 6 above)
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/src/avaxp.ts:105]: The addresses were initially missing the "P-". I think that it changes. I'll check if I could improve it,

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts (hunk 8 above)
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/test/unit/avaxp.ts:345]: done

File: modules/sdk-coin-avaxp/src/avaxp.ts (hunk 7 above)
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/src/avaxp.ts:101]: done

File: modules/sdk-coin-avaxp/src/avaxp.ts (hunk 6 above)
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/src/avaxp.ts:105]: done

File: modules/sdk-coin-avaxp/src/avaxp.ts (hunk 5 above)
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/src/avaxp.ts:91]: done

**Review by CapnMigraine — APPROVED** [review]

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**